gravitee-apim-rest-api/gravitee-apim-rest-api-idp/gravitee-apim-rest-api-idp-core/src/main/java/io/gravitee/rest/api/idp/core/plugin/impl/IdentityProviderManagerImpl.java

```diff
@@ -129,6 +130,8 @@ public ConfigurableEnvironment environment() {
                             protected void customizePropertySources(MutablePropertySources propertySources) {
                                 propertySources.addFirst(new RelaxedPropertySource(plugin.id(), properties));
                                 super.customizePropertySources(propertySources);
+                                this.getConversionService().addConverter(Secret.class, byte[].class, Secret::asBytes);
+                                this.getConversionService().addConverter(Secret.class, String.class, Secret::asString);
```

Comment:
Should we reuse the same logic we have in `GraviteeConfigurationSecretResolverDispatcher` to make sure it covers all the cases? ie:

```java
private void setupConverters(ConfigurableEnvironment environment) {
    environment
        .getConversionService()
        .addConverterFactory(
            new ConverterFactory<Secret, Object>() {
                final ConversionService conversionService = DefaultConversionService.getSharedInstance();

                @Nonnull
                public <T> Converter<Secret, T> getConverter(@Nonnull Class<T> targetType) {
                    return source -> conversionService.convert(source.asString(), targetType);
                }
            }
        );

    environment.getConversionService().addConverter(Secret.class, byte[].class, Secret::asBytes);
}
```

Reply:
This is not what is done in Gravitee Node, Gravitee node uses the last line syntax of your example

Reply:
(well to be accurate, it uses another method signature, but that's just details)

Reply:
Yeah I copy/paste to give the idea, obviously it deserve adaptations but they seems to be really trivial ;-)